Index: generic/tkImgPhoto.c
===================================================================
RCS file: /cvsroot/tktoolkit/tk/generic/tkImgPhoto.c,v
retrieving revision 1.36.2.12
diff -u -r1.36.2.12 tkImgPhoto.c
--- generic/tkImgPhoto.c	3 Dec 2004 13:16:35 -0000	1.36.2.12
+++ generic/tkImgPhoto.c	9 Dec 2004 09:58:22 -0000
@@ -4586,7 +4586,9 @@
      * Check if display code needs alpha blending...
      */
 
-    ToggleComplexAlphaIfNeeded(masterPtr);
+    if (alphaOffset != 0 || masterPtr->flags & COMPLEX_ALPHA) {
+	ToggleComplexAlphaIfNeeded(masterPtr);
+    }
 
     /*
      * Update each instance.
@@ -4878,7 +4880,9 @@
      * Check if display code needs alpha blending...
      */
 
-    ToggleComplexAlphaIfNeeded(masterPtr);
+    if (alphaOffset != 0 || masterPtr->flags & COMPLEX_ALPHA) {
+	ToggleComplexAlphaIfNeeded(masterPtr);
+    }
 
     /*
      * Update each instance.
