*** tclVar.c.orig	Fri Nov 21 11:46:38 2003
--- tclVar.c	Fri Nov 21 11:46:42 2003
***************
*** 616,622 ****
   *     namespace; never follow the second (global) resolution path 
   *   - Bug #631741 - do not use special namespace or interp resolvers
   */
! #define LOOKUP_FOR_UPVAR 0x400
  
  /*
   *----------------------------------------------------------------------
--- 616,627 ----
   *     namespace; never follow the second (global) resolution path 
   *   - Bug #631741 - do not use special namespace or interp resolvers
   */
! /* As reported in Bug #835020: define LOOKUP_FOR_UPVAR to 0x40000 to avoid
!  * a collision with the deprecated TCL_PARSE_PART1, used throughout TclX
!  * EAL 2003-11-21
!  * #define LOOKUP_FOR_UPVAR 0x400
!  */
! #define LOOKUP_FOR_UPVAR 0x40000
  
  /*
   *----------------------------------------------------------------------
