*** tcl.m4.orig	Tue Mar  6 08:14:46 2007
--- tcl.m4	Mon Sep 10 12:56:11 2007
***************
*** 773,779 ****
  	    # results, and the version is kept in special file).
  	
  	    if test -r /etc/.relid -a "X`uname -n`" = "X`uname -s`" ; then
! 		system=MP-RAS-`awk '{print $3}' /etc/.relid'`
  	    fi
  	    if test "`uname -s`" = "AIX" ; then
  		system=AIX-`uname -v`.`uname -r`
--- 773,779 ----
  	    # results, and the version is kept in special file).
  	
  	    if test -r /etc/.relid -a "X`uname -n`" = "X`uname -s`" ; then
! 		system=MP-RAS-`awk '{print $3}' /etc/.relid`
  	    fi
  	    if test "`uname -s`" = "AIX" ; then
  		system=AIX-`uname -v`.`uname -r`
***************
*** 2148,2154 ****
  	    # results, and the version is kept in special file).
  	
  	    if test -r /etc/.relid -a "X`uname -n`" = "X`uname -s`" ; then
! 		system=MP-RAS-`awk '{print $3}' /etc/.relid'`
  	    fi
  	    if test "`uname -s`" = "AIX" ; then
  		system=AIX-`uname -v`.`uname -r`
--- 2148,2154 ----
  	    # results, and the version is kept in special file).
  	
  	    if test -r /etc/.relid -a "X`uname -n`" = "X`uname -s`" ; then
! 		system=MP-RAS-`awk '{print $3}' /etc/.relid`
  	    fi
  	    if test "`uname -s`" = "AIX" ; then
  		system=AIX-`uname -v`.`uname -r`
