*** configure.orig	Tue Mar  6 08:14:45 2007
--- configure	Mon Sep 10 12:56:07 2007
***************
*** 7247,7253 ****
  	    # results, and the version is kept in special file).
  
  	    if test -r /etc/.relid -a "X`uname -n`" = "X`uname -s`" ; then
! 		system=MP-RAS-`awk '{print }' /etc/.relid'`
  	    fi
  	    if test "`uname -s`" = "AIX" ; then
  		system=AIX-`uname -v`.`uname -r`
--- 7247,7253 ----
  	    # results, and the version is kept in special file).
  
  	    if test -r /etc/.relid -a "X`uname -n`" = "X`uname -s`" ; then
! 		system=MP-RAS-`awk '{print }' /etc/.relid`
  	    fi
  	    if test "`uname -s`" = "AIX" ; then
  		system=AIX-`uname -v`.`uname -r`
